File: backend/auth-service/src/main/java/com/example/authservice/oauth/grant/admin/AdminPasswordGrantToken.java
package com.example.authservice.oauth.grant.admin;

import com.example.authservice.oauth.config.AuthorizationServerConfig;
import java.util.Map;
import java.util.Set;
import org.springframework.security.core.Authentication;
import org.springframework.security.oauth2.server.authorization.authentication.OAuth2AuthorizationGrantAuthenticationToken;

public class AdminPasswordGrantToken extends OAuth2AuthorizationGrantAuthenticationToken {

    private final String email;
    private final String password;
    private final Set<String> scopes;

    public AdminPasswordGrantToken(Authentication clientPrincipal, String email, String password, Set<String> scopes, Map<String, Object> additionalParameters) {
        super(AuthorizationServerConfig.ADMIN_PASSWORD, clientPrincipal, additionalParameters);
        this.email = email;
        this.password = password;
        this.scopes = scopes;
    }

    public String getEmail() { return email; }
    public String getPassword() { return password; }
    public Set<String> getScopes() { return scopes; }
}

File: backend/auth-service/src/main/kotlin/com/example/authservice/oauth/grant/admin/AdminPasswordGrantToken.kt
package com.example.authservice.oauth.grant.admin

import com.example.authservice.oauth.config.AuthorizationServerConfig
import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.server.authorization.authentication.OAuth2AuthorizationGrantAuthenticationToken

class AdminPasswordGrantToken(
    clientPrincipal: Authentication,
    val email: String,
    val password: String,
    val scopes: Set<String>,
    additionalParameters: Map<String, Any>,
) : OAuth2AuthorizationGrantAuthenticationToken(AuthorizationServerConfig.ADMIN_PASSWORD, clientPrincipal, additionalParameters)
